includes/Checker/Abstract_Check_Runner.php

```diff
@@ -194,6 +211,24 @@ final public function set_experimental_flag( $include_experimental ) {
 		$this->include_experimental = $include_experimental;
 	}
 
+	/**
+	 * Sets categories for filtering the checks.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $categories An array of categories for filtering.
+	 *
+	 * @throws Exception Thrown if the getegories does not match the original request parameter.
+	 */
+	final public function filter_checks_by_specific_categories( $categories ) {
+		if ( $categories !== $this->get_categories_param() ) {
+			throw new Exception(
+				__( 'Invalid flag: The include-experimental flag does not match the original request parameter.', 'plugin-check' )
```

Comment:
I'd recommend using `sprintf()` here for `include-experimental` so that translators don't have to translate this piece.

```php
sprintf(
	/* translators: %s: include-experimental */
	__( 'Invalid flag: The %s flag does not match the original request parameter.', 'plugin-check' ),
	'include-experimental'
)
```

Reply:
Thanks @swissspidy, Updated in https://github.com/10up/plugin-check/pull/228/commits/5c679dc04e4ae6259a95ec6f07ddfb7a6fe6efc5